### PR TITLE
ci: Use macos-14 GHA image (x86_64-apple-darwin22.6.0 -> arm64-apple-darwin23.6.0)

### PR DESCRIPTION
**Pull request by maflcko**
There shouldn't be any downside, because XCode remains pinned to the same version.

However, builds are expected to be a bit faster with M1, which seems nice.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -74,11 +74,11 @@ jobs:
           # Use clang++, because it is a bit faster and uses less memory than g++
           git rebase --exec "echo Running test-one-commit on \$( git log -1 ) && CC=clang CXX=clang++ cmake -B build -DWITH_ZMQ=ON -DBUILD_GUI=ON -DBUILD_BENCH=ON -DBUILD_FUZZ_BINARY=ON -DWITH_BDB=ON -DWITH_NATPMP=ON -DWITH_MINIUPNPC=ON -DWITH_USDT=ON && cmake --build build -j $(nproc) && ctest --test-dir build -j $(nproc) && ./build/test/functional/test_runner.py -j $(( $(nproc) * 2 ))" ${{ env.TEST_BASE }}
 
-  macos-native-x86_64:
-    name: 'macOS 13 native, x86_64, no depends, sqlite only, gui'
+  macos-native-arm64:
```
Comment (willcl-ark) [inline .github/workflows/ci.yml:77]: This is still building using x86_64-apple-darwin as the host (from 00_setup_env_mac_native.sh). I wonder if the "native" here is therefore confusing?

**Review by willcl-ark — COMMENTED** [review]
Concept ACK.

Very much in favour of speeding these jobs up! Changes look good to me so far, just one Q on the jobname with this change in architecture.

**Review by hebasto — COMMENTED** [review]
Concept ACK.

Perhaps explicitly mention the switch from the `x86_64` architecture to `arm64`?

**Review by hebasto — APPROVED** [review]
ACK fab932b4211ea0a4c10e484c08042552f9c58c15.

We were already using an `arm64` CI task before https://github.com/bitcoin/bitcoin/pull/28187.

**Review by willcl-ark — APPROVED** [review]
ACK fab932b4211ea0a4c10e484c08042552f9c58c15

**Pull request merged**